Makd.mak

```diff
@@ -565,6 +565,40 @@ $O/test-%.stamp: $O/test-%
 	$(call exec,$< $(ITFLAGS),$<,run)
 	$Vtouch $@
 
+# examples rules
+##########################
+
+# examples are assumed to be standalone programs, so we recursively search
+# for files examples/*.d and assume that each file is a separate example
+# program. Files can exist in sub-directories under examples/ dir.
+# The sources list is filtered through the $(EXAMPLES_FILTER_OUT) variable 
+# contents (using the Make function $(filter-out)), so you can exclude an
+# example by adding the example file path (as an absolute path using $C) to
+# this variable.
+
+# The 'example' target builds all the examples.
+.PHONY: example
+example: $(patsubst $T/example/%.d,$O/example/%,\
+	$(filter-out $(EXAMPLE_FILTER_OUT),$(shell find $T/example/ -name "*.d")))
```

Comment:
This `find` command is definitely incorrect, at least if we want to allow examples that consist on more than one file. See how `integrationtests` are done. My suggestion is to support both:

1. Top-level `<name>.d` files for simple one-file examples
2. More complex examples that need multiple files, that should be in `<name>/main.d`

(I would extend this to `integrationtests` too, since in most cases tests are only one file and the extra depth is just annoying)